File: BE/src/main/java/com/manager/account/interfaces/rest/dto/response/UserResponse.java
package com.manager.account.interfaces.rest.dto.response;

import lombok.Getter;
import lombok.Setter;

import java.time.LocalDateTime;


@Setter
@Getter
public class UserResponse {
    private String employeeId;
    private String displayName;
    private String role;
    private LocalDateTime createdAt;
    private String createdBy;

    public UserResponse(String employeeId, String displayName, String role, LocalDateTime createdAt, String createdBy) {
        this.employeeId = employeeId;
        this.displayName = displayName;
        this.role = role;
        this.createdAt = createdAt;
        this.createdBy = createdBy;
    }
}

File: FE/app/src/main/java/com/food/order/data/response/UserResponse.kt
package com.food.order.data.response

import com.food.order.data.model.UserModel
import com.google.gson.annotations.SerializedName
import java.io.Serializable

data class UserResponse(
    @SerializedName("uid") val uid: String?,
    @SerializedName("fullName") val fullName: String?,
    @SerializedName("email") val email: String?,
    @SerializedName("phone_number") val phoneNumber: String?,
    @SerializedName("role") val role: String?,
    @SerializedName("server") val server: String?,
    @SerializedName("birthday") val birthday: String?,
    @SerializedName("gender") val gender: String?,
    @SerializedName("createdDate") val createdDate: String?,
    @SerializedName("created_by") val createdBy: String?
) : Serializable {
    fun toUserModel(): UserModel {
        return UserModel(
            employeeId = uid ?: "UNKNOWN_ID",
            displayName = fullName ?: "No Name",
            role = role ?: "USER",
            server = server ?: "HCM",
            birthday = birthday,
            email = email,
            phoneNumber = phoneNumber,
            gender = gender,
            createdAt = createdDate,
            createdBy = createdBy
        )
    }
}
